pkg/mads/server/server_test.go

```diff
@@ -0,0 +1,133 @@
+package server_test
+
+import (
+	"bytes"
+	"net/http"
+
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/golang/protobuf/jsonpb"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/common/config"
+
+	kuma_cp "github.com/kumahq/kuma/pkg/config/app/kuma-cp"
+	"github.com/kumahq/kuma/pkg/config/mads"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/runtime"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/mads/server"
+	mads_v1_client "github.com/kumahq/kuma/pkg/mads/v1/client"
+	"github.com/kumahq/kuma/pkg/metrics"
+	"github.com/kumahq/kuma/pkg/plugins/resources/memory"
+)
+
+type testRuntime struct {
+	runtime.Runtime
+	rm         manager.ResourceManager
+	config     kuma_cp.Config
+	components []component.Component
+	metrics    metrics.Metrics
+}
+
+func (t *testRuntime) ReadOnlyResourceManager() manager.ReadOnlyResourceManager {
+	return t.rm
+}
+
+func (t *testRuntime) Add(component ...component.Component) error {
+	t.components = append(t.components, component...)
+	return nil
+}
+
+func (t *testRuntime) Config() kuma_cp.Config {
+	return t.config
+}
+
+func (t *testRuntime) Metrics() metrics.Metrics {
+	return t.metrics
+}
+
+var _ = Describe("MADS Server", func() {
+
+	var rt *testRuntime
+	var stopCh chan struct{}
+	var errCh chan error
+
+	BeforeEach(func() {
+		m, err := metrics.NewMetrics("zone-1")
+		Expect(err).ToNot(HaveOccurred())
+
+		rt = &testRuntime{
+			rm:      manager.NewResourceManager(memory.NewStore()),
+			config:  kuma_cp.Config{MonitoringAssignmentServer: mads.DefaultMonitoringAssignmentServerConfig()},
+			metrics: m,
+		}
+		rt.config.MonitoringAssignmentServer.Port = 15676
```

Comment:
no port hardcoding in tests! please randomize the port

Reply:
Replaced with`PickTCPPort`, is it okay?

Reply:
it is